src/streaming/TextSourceBuffer.js

```diff
@@ -91,37 +138,33 @@ MediaPlayer.dependencies.TextSourceBuffer = function () {
                 bytes = new Uint8Array(bytes);
                 ccContent=window.UTF8.decode(bytes);
                 try {
-                    result = self.getParser().parse(ccContent);
-                    label = mediaInfo.id;
-                    lang = mediaInfo.lang;
-                    self.getTextTrackExtensions().addTextTrack(self.videoModel.getElement(), result, label, lang, true);
-                    self.eventBus.dispatchEvent({type:MediaPlayer.events.TEXT_TRACK_ADDED});
+                    result = self.getParser(mimeType).parse(ccContent);
+                    createTextTrackFromMediaInfo(result, mediaInfo);
                 } catch(e) {
                     self.errHandler.closedCaptionsError(e, "parse", ccContent);
                 }
             }
         },
 
+        getIsDefault:function(mediaInfo){
+            return mediaInfo.index === this.mediaInfos[0].index; //TODO How to tag default. currently same order as in manifest. Is there a way to mark a text adaptation set as the default one?
+        },
+
         abort:function() {
-            this.getTextTrackExtensions().deleteCues(this.videoModel.getElement());
+            this.videoModel.getElement().textTracks.removeEventListener('change', onTextTrackChange);
+            this.textTrackExtensions.deleteAllTextTracks();
         },
 
-        getParser:function() {
+        getParser:function(mimeType) {
             var parser;
-
             if (mimeType === "text/vtt") {
                 parser = this.system.getObject("vttParser");
-            } else if (mimeType === "application/ttml+xml") {
+            } else if (mimeType === "application/ttml+xml" || "fragmentedText") {
```

Comment:
Let me clean this up.   Put in way before the discuss popped up about mimeType and fragmented text.   Good catch.